### DISCUSSION TITLE
Questions to run outline on local server in dev mode

### DISCUSSION THREAD

**Discussion by zfrzhangfurui** [outline/outline#1618] [Self Hosting] [answered]
Hi everyone,
  I am a new coder and have some questions with how to run outline in develop mode on my dock container. 
1. In readme, development section. what does it mean to Copy the file .env.sample to .env. Is that mean create a new .env file and copy the code? or put it in the env.js file in server folder?  (I have a little bit experience with angular, in angular there are two env files in root folder) 
2. Is that possible to run outline backend and database on my local server but not use AWS? 

Kind Regards
Furui Zhang

**Comment by tommoor** [comment, marked as answer]
> Is that mean create a new .env file and copy the code?

Yes, or just copy the file – It should be in the same folder.

> Is that possible to run outline backend and database on my local server but not use AWS?

Yes, you can use any S3-compatible service. If you want to run it locally a lot of users have had success using a [Minio](https://github.com/minio/minio) container.

**Reply by zfrzhangfurui** [reply]
Thanks a lot!

**Comment by MDXDave** [comment]
Take a look at https://github.com/chsasank/outline-wiki-docker-compose
It made it really simple to run an outline instance.

**Reply by zfrzhangfurui** [reply]
Thanks a lot !
